core/src/test/java/de/evoila/cf/broker/controller/core/CatalogControllerTest.java

```diff
@@ -0,0 +1,136 @@
+     * definitions.
+     * @param filePath  The path (relative or absolute) to the JSON file.
+     * @return  A list containing all service definitions contained in the submitted file.
+     * @throws Exception
+     */
+    private List<ServiceDefinition> readServicesFile(String filePath) throws Exception
+    {
+        List<ServiceDefinition> generatedServices;
+        try (FileReader fileReader = new FileReader(new File(filePath))) {
+            generatedServices = new Gson().fromJson(fileReader, new TypeToken<List<ServiceDefinition>>() {}.getType());
+        }
+        return generatedServices;
+    }
+
+    /**
+     * A function testing CatalogController.getCatalog with a valid catalog.
+     * The catalog is stored in a JSON file.
+     * Not an @Test but called in @Test methods.
+     * @param catalogPath           The path of the JSON file containing the catalog.
+     * @param requestIdentity       A string representing a request identity.
+     * @param originatingIdentity   A string representing a originating identity.
+     * @throws Exception
+     */
+    private void validCatalogTest(String catalogPath,
+                                  String requestIdentity,
+                                  String originatingIdentity) throws Exception
+    {
+        // mocks
+        when(catalogService.getCatalog()).thenReturn(catalog);
+
+        List<ServiceDefinition> generatedServices = readServicesFile(catalogPath);
+        when(catalog.getServices()).thenReturn(generatedServices);
+        // actual method call
+        ResponseEntity<Catalog> response = catalogController.getCatalog(requestIdentity,
+                originatingIdentity);
+        // create expected object
+        Catalog expectedCatalog = new Catalog();
+        List<ServiceDefinition> expectedServices = readServicesFile(catalogPath);
+        expectedCatalog.setServices(expectedServices);
+        // check results
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(expectedCatalog, response.getBody()); // ist dieser Test wirklich sinnvoll? Wir mocken ja gerade getServices. Wäre es ausreichend zu prüfen, dass der Catalog nicht null ist? Oder nicht leer? In diesen Fällen bräuchte man auch die ganzen equals Funktionen nicht hinzufügen.
+    }
+
+    @Test
+    void getCatalog_EmptyCatalog() {
+        // mocks
+        when(catalogService.getCatalog()).thenReturn(catalog);
+        when(catalog.getServices()).thenReturn(new ArrayList<>());
+        // actual method call
+        ResponseEntity<Catalog> response = catalogController.getCatalog(REQUEST_IDENTITY,
+                                                                        ORIGINATING_IDENTITY);
+        // check results
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(new Catalog(), response.getBody());
+    }
+
+    @Test
+    void getCatalog_ValidCatalog()
```

Comment:
Please do not use underscores in method names. In Java the camel case notation is commonly used for naming.